spec/models/journaled/writer_spec.rb

```diff
@@ -208,5 +208,112 @@
         end
       end
     end
+
+    context 'when inside of a transaction' do
+      def fake_event(num)
+        instance_double(
+          event_class,
+          journaled_schema_name: 'fake_schema_name',
+          journaled_attributes: { id: "FAKE_UUID_#{num}", event_type: "fake_event_#{num}", created_at: Time.zone.now, foo: :bar },
+          journaled_partition_key: 'fake_partition_key',
+          journaled_stream_name: 'my_app_events',
+          journaled_enqueue_opts: journaled_enqueue_opts,
+          tagged?: false,
+        )
+      end
+
+      it 'batches multiple events and does not enqueue until the end of a transaction' do
+        expect {
+          ActiveRecord::Base.transaction do
+            expect { described_class.new(journaled_event: fake_event(1)).journal! }
+              .to not_change { enqueued_jobs.count }.from(0)
+              .and not_journal_event_including(id: 'FAKE_UUID_1', event_type: 'fake_event_1')
+            expect { described_class.new(journaled_event: fake_event(2)).journal! }
+              .to not_change { enqueued_jobs.count }.from(0)
+              .and not_journal_event_including(id: 'FAKE_UUID_2', event_type: 'fake_event_2')
+          end
+        }.to change { enqueued_jobs.count }.from(0).to(1)
+          .and journal_event_including(id: 'FAKE_UUID_1', event_type: 'fake_event_1')
+          .and journal_event_including(id: 'FAKE_UUID_2', event_type: 'fake_event_2')
+      end
+
+      context 'when a transaction rolls back' do
+        it 'batches multiple events and does not enqueue until the end of a transaction' do
+          expect {
+            ActiveRecord::Base.transaction do
+              expect { described_class.new(journaled_event: fake_event(1)).journal! }
+                .to not_change { enqueued_jobs.count }.from(0)
+                .and not_journal_event_including(id: 'FAKE_UUID_1', event_type: 'fake_event_1')
+              expect { described_class.new(journaled_event: fake_event(2)).journal! }
+                .to not_change { enqueued_jobs.count }.from(0)
+                .and not_journal_event_including(id: 'FAKE_UUID_2', event_type: 'fake_event_2')
+              raise ActiveRecord::Rollback
+            end
+          }.to not_change { enqueued_jobs.count }.from(0)
+            .and not_journal_event_including(id: 'FAKE_UUID_1', event_type: 'fake_event_1')
+            .and not_journal_event_including(id: 'FAKE_UUID_2', event_type: 'fake_event_2')
+
+          # Make sure that prior transaction does not impact behavior of future transactions:
```

Comment:
We just want to make sure that the commit/rollback (`active?` check) properly gives us a fresh `TransactionHandler` in the connection extension code. This might be a little defensive but it was an easy enough case to throw in.